### PR TITLE
New templates and changes to existing templates

### PR DESCRIPTION
**Pull request by waptik**
## About
This PR introduces a bunch of new templates & components that we'll be releasing soon and changes to existing templates

### New Templates & Component
- [x] RSS Frame 
- [x] Meme Creator Frame
- [x] Fundraiser Frame
- [x] Token Swap Frame
- [x] Gated Frame
- [x] Smart Contract Frame
- [x] Gating component
- [x] TextSlide component

<details><summary>Details about new templates</summary>
<p>

## Meme Creator Frame
The template allows you to create a meme frame based on popular meme images and add texts in the right places (see #90 )
https://www.loom.com/share/d4c49a41faa24fe78e67906405f8c903

## RSS Frame
This template allows you to create a frame that converts any public rss feed into a series of digestible posts right inside it
https://www.loom.com/share/e52b3847055847348350b842d66607c2

## Fundraiser Frame (#108)
This template allows Users to create donation/fundraise frame with ease.
The demo video below shows how it works from both the User's and Viewer's perspective. The success screen didn't show up because of slow internet connection
https://www.loom.com/share/3c5657ed23f94523917a6fab98f17090

## Gated Frame (#109)
This template allows users to create frames that'll reward their audience that fall within certain categories by using gate keeping to filter them out. Approved audience will then be able to see their rewards which come in the form of nft images or external links

## Smart Contract Frame (#110)
This template allows Users to create a frame for any smart contract on the chains FT supports, allowing Viewers to navigate through each available functions and query them.
The demo video below shows how it works. Due to slow internet connectivity whilst recording the video, there's been lots of time outs and i had to stop recording midway.
https://www.loom.com/share/859b8b249ff343cf997e10ce2a30aefd

## TextSlide component
The `TextSlide` is a view component that can be used in any template's handler that needs to display a text view where at least one text element is needed and at most 3 text elements.
It's composed of the following:
- Title(required): the main text to be displayed in a slide 
- Subtitle(optional): this is more like a description text element
- Custom message(optional): an extra text element which serves as a notice

## Summary by CodeRabbit

- **New Features**
	- Enhanced integration capabilities with new configuration variables for various APIs and services.
	- New API endpoint for efficiently managing CDN-related files.
	- Functionality for interacting with the 0x Swap API, enabling price and swap quote retrieval.
	- Dynamic handling of transaction success scenarios with improved user interface components.
	- Introduction of a structured fundraising interface with customizable settings.
	- New template for creating gated experiences that reward user engagement.
	- Enhanced form management with new properties for user engagement metrics.
	- New module for handling RSS feeds with flexible configuration options.

- **Bug Fixes**
	- Streamlined error handling and response structures in various components.
	- Improvements in the logic and user interface of existing components for better usability.

- **Documentation**
	- Updates to configuration documentation reflecting new variables and components.

</p>
</details> 

### Changes to existing Templates

- [x] Add gating to `Poll` & `Form`
- [x] Refactor `Form` template + enhancements
- [x] Added timezone, badges to `Cal`
- [x] Fix background issues in `Presentation` views
- [x] fix emoji support Composer Actions

### CODE REVIEW

File: app/api/vacuum/route.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            },
+        })
+        const objects = await s3.listObjects({ Bucket: `${process.env.S3_BUCKET}` })
+        const files = (objects.Contents ?? []).map((object) => object.Key) as string[]
+        const foundFiles: string[] = []
+
+        console.log(`Found ${files.length} files in S3 and ${frames.length} frames in the database`)
+
+        for (const frame of frames) {
+            const config = frame.config ?? {}
```
Comment (FTCHD) [inline app/api/vacuum/route.ts:29]: Collect paths from `draftConfig` too.

File: app/api/vacuum/route.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            const urls = collectFilePaths(config)
+            foundFiles.push(...urls)
+        }
+
+        const filesToDelete = files.filter((file) => !foundFiles.includes(file))
+
+        await deleteFilesFromR2(s3, filesToDelete)
+
+        return Response.json({
```
Comment (FTCHD) [inline app/api/vacuum/route.ts:38]: Return only the length for each (files, filesToDelete, filesFound).

File: components/home/TemplateCard.tsx [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { createFrame } from '@/lib/frame'
 import type { getTemplates } from '@/lib/template'
 import type templates from '@/templates'
 import { useSession } from 'next-auth/react'
+import type { StaticImageData } from 'next/image'
```
Comment (FTCHD) [inline components/home/TemplateCard.tsx:6]: This type is not used throughout the file.

File: lib/serve.ts [hunk 4]
```diff
@@ -25,7 +25,7 @@ export async function buildFramePage({
     id: string
     linkedPage: string | undefined
 } & BuildFrameData) {
-    if (!component && !image) {
+    if (!(component || image)) {
```
Comment (FTCHD) [inline lib/serve.ts:28]: Revert.

File: app/api/vacuum/route.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            files: files.length,
+            filesToDelete: filesToDelete.length,
+            foundFiles: foundFiles.length,
```
Comment (FTCHD) [inline app/api/vacuum/route.ts:41]: `filesFound`

File: app/api/vacuum/route.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        })
+        const objects = await s3.listObjects({ Bucket: `${process.env.S3_BUCKET}` })
+        const files = (objects.Contents ?? []).map((object) => object.Key) as string[]
+        const foundFiles: string[] = []
+
+        console.log(`Found ${files.length} files in R2 and ${frames.length} frames in the database`)
+
+        for (const frame of frames) {
+            const draftConfig = frame.draftConfig ?? {}
+            const urls = collectFilePaths(draftConfig)
```
Comment (FTCHD) [inline app/api/vacuum/route.ts:30]: Collect URLs from both `draftConfig` and `config`. 

To be safe in the future, collect URLs from `storage` too.

File: .env.example [hunk 7]
```diff
@@ -17,5 +17,21 @@ S3_REGION=auto
 S3_KEY_ID=
 S3_SECRET=
 
-IMGFLIG_USERNAME=
-IMGFLIG_PASSWORD=
\ No newline at end of file
+# 0x.org api key
+ZEROX_API_KEY=
```
Comment (FTCHD) [inline .env.example:21]: ZEROEX is the usual spelling.

Comment (waptik) [reply .env.example:21]: as prefix or env var?

Comment (FTCHD) [reply .env.example:21]: ZEROEX_API_KEY

File: app/api/vacuum/route.ts [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        const files = (objects.Contents ?? []).map((object) => object.Key) as string[]
+        const filesFound: string[] = []
+
+        console.log(`Found ${files.length} files in R2 and ${frames.length} frames in the database`)
+
+        for (const frame of frames) {
+            const draftConfig = frame.draftConfig ?? {}
+            const config = frame.config ?? {}
+            const storage = frame.storage ?? {}
+            const paths = collectFilePaths([draftConfig, config, storage])
```
Comment (FTCHD) [inline app/api/vacuum/route.ts:32]: Make sure this is right and you don't need to do `collectFilePaths([...draftConfig, ...config, ...storage])`.

The `traverse` function probably handles it correctly, but double check.

Comment (waptik) [reply app/api/vacuum/route.ts:32]: This is from my own instance(iuse the same bucket for vercel instance and local repo). Screenshot from local repo with local sqlite db
image
image

**Pull request merged**